src/arkode/arkode_lsrkstep_io.c

```diff
@@ -0,0 +1,661 @@
+  LSRKStepSetDomEigFn specifies the dom_eig function.
+  ---------------------------------------------------------------*/
+int LSRKStepSetDomEigFn(void* arkode_mem, ARKDomEigFn dom_eig)
+{
+  ARKodeMem ark_mem;
+  ARKodeLSRKStepMem step_mem;
+  int retval;
+
+  /* access ARKodeMem and ARKodeLSRKStepMem structures */
+  retval = lsrkStep_AccessARKODEStepMem(arkode_mem, __func__, &ark_mem,
+                                        &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  /* set the dom_eig routine pointer, and update relevant flags */
+  if (dom_eig != NULL)
+  {
+    step_mem->is_ext_dom_eig = SUNTRUE;
+    step_mem->extDomEig      = dom_eig;
+
+    return (ARK_SUCCESS);
+  }
+  else
+  {
+    step_mem->is_ext_dom_eig = SUNFALSE;
+    step_mem->extDomEig      = NULL;
+
+    arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                    "Internal dom_eig is not supported yet!");
+    return (ARK_ILL_INPUT);
+  }
+}
+
+/*---------------------------------------------------------------
+  LSRKStepSetDomEigFrequency sets dom_eig computation frequency -
+  Dominated Eigenvalue is recomputed after "nsteps" successful steps.
+
+    nsteps = 0 refers to Constant Jacobian
+  ---------------------------------------------------------------*/
+int LSRKStepSetDomEigFrequency(void* arkode_mem, int nsteps)
+{
+  ARKodeMem ark_mem;
+  ARKodeLSRKStepMem step_mem;
+  int retval;
+
+  /* access ARKodeMem and ARKodeLSRKStepMem structures */
+  retval = lsrkStep_AccessARKODEStepMem(arkode_mem, __func__, &ark_mem,
+                                        &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  if (nsteps < 0)
+  {
+    arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                    "nsteps must be greater than or equal to 0");
+    return (ARK_ILL_INPUT);
+  }
+
+  if (nsteps == 0)
+  {
+    step_mem->const_Jac    = SUNTRUE;
+    step_mem->dom_eig_freq = 1;
+  }
+  else
+  {
+    step_mem->dom_eig_freq = nsteps;
+    step_mem->const_Jac    = SUNFALSE;
+  }
+
+  return (ARK_SUCCESS);
+}
+
+/*---------------------------------------------------------------
+  LSRKStepSetMaxNumStages sets the maximum number of stages allowed.
+  ---------------------------------------------------------------*/
+int LSRKStepSetMaxNumStages(void* arkode_mem, int stage_max_limit)
+{
+  ARKodeMem ark_mem;
+  ARKodeLSRKStepMem step_mem;
+  int retval;
+
+  /* access ARKodeMem and ARKodeLSRKStepMem structures */
+  retval = lsrkStep_AccessARKODEStepMem(arkode_mem, __func__, &ark_mem,
+                                        &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  if (stage_max_limit < 2)
+  {
+    arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                    "stage_max_limit must be greater than or equal to 2");
+    return (ARK_ILL_INPUT);
+  }
+
+  step_mem->stage_max_limit = stage_max_limit;
```

Comment:
I see this has a default value based on the tolerance. Should there be a default max and way to reset this to the default value? I also don't see where this is adjusted when the user sets their tolerance. Setting this value should be deferred to `lsrkStep_Init` at that point the tolerances will have been set.